src/js/utils/AutoDetect.js

```diff
@@ -0,0 +1,176 @@
+import PortHandler from '../port_handler';
+import { gui_log } from '../gui_log';
+import { i18n } from "../localization";
+import { TABS } from '../gui';
+import MspHelper from '../msp/MSPHelper';
+import FC from '../fc';
+import MSP from '../msp';
+import MSPCodes from '../msp/MSPCodes';
+import semver from 'semver';
+import { API_VERSION_1_39, API_VERSION_1_45, API_VERSION_1_46 } from '../data_storage';
+import serial from '../webSerial';
+
+/**
+ *
+ * Auto-detect firmware flashed and return target
+ *
+ */
+
+let mspHelper = null;
+
+function ReadSerialAdapter(event) {
```

Comment:
```suggestion
function readSerialAdapter(event) {
```

Reply:
One question more. Why is this out of the class? Maybe to have something like a "private" method?

Reply:
It's a helper function and does not interact with the class directly.